scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/GoalBuilder.java

```diff
@@ -135,8 +161,16 @@ private static Expression<Windows> expressionOfConstraintExpression(
   private static ActivityCreationTemplate makeActivityTemplate(
       final SchedulingDSL.ActivityTemplate activityTemplate,
       final Function<String, ActivityType> lookupActivityType) {
-    var builder = new ActivityCreationTemplate.Builder()
-        .ofType(lookupActivityType.apply(activityTemplate.activityType()));
+    var builder = new ActivityCreationTemplate.Builder();
+    final var type = lookupActivityType.apply(activityTemplate.activityType());
+    if(type.getDurationType() instanceof DurationType.Controllable durationType){
+      //detect duration parameter
+      if(activityTemplate.arguments().containsKey(durationType.parameterName())){
+        builder.duration(new DurationValueMapper().deserializeValue(activityTemplate.arguments().get(durationType.parameterName())).getSuccessOrThrow());
```

Comment:
This is the best we can do for now - however, I think we'll run into issues with this soon, because users can (and do) use different value mappers than what we provide. (In particular, Clipper overrides the `DurationValueMapper` to return more human-readable units, instead of microseconds as a long)

I've captured this in a ticket:
> AERIE-1863: Do not assume Durations are serialized as integers
> 
> In order to interpret the expected duration of a ControllableDuration activity, the scheduler must be able to deserialize a serialized Duration argument into a Duration java object.
> 
> Mission models are allowed to override the default value mappers we provide them, so it is not safe for us to assume anything about the serialized representation of Durations.
> 
> One solution is to add methods to the SchedulerModel that will serialize and deserialize Durations for us. The correct value mapper will be available at annotation processing time, so the generated SchedulerModel should have access to it.